src/reports/models/reportsManager.js

```diff
@@ -14,13 +14,17 @@ module.exports.getReport = async (testId, reportId) => {
         throw error;
     }
 
-    let report = getReportResponse(reportSummary[0]);
+    let report = await getReportResponse(reportSummary[0]);
     return report;
 };
 
 module.exports.getReports = async (testId) => {
     let reportSummaries = await databaseConnector.getReports(testId);
-    let reports = reportSummaries.map(getReportResponse);
+    let reports = [];
+    for (let i = 0; i < reportSummaries.length; i++) {
+        const reportResponse = await getReportResponse(reportSummaries[i]);
+        reports.push(reportResponse);
+    }
     return reports;
```

Comment:
for each summary you make request from the db.
do one request to the db , before the loop and move the arguments as config object to getReportResponse
and getReportResponse will not be async anymore.

Reply:
👍